tests/hm_typing/simple_generics.sy

```diff
@@ -0,0 +1,14 @@
+A :: enum(*A, *B)
+    C *A,
+    D *B,
+end
+
+B :: blob(*A, *B) {
+    a: *A,
+    b: *B,
+}
+
+start :: fn do
+    a : A(int, int) : A.C 1
+    b : B(int) : B { a: 1, b: 2 }
```

Comment:
it seems like both A and B is int here. what happens if i specify 2 out of 3 generics?

Reply:
reasonable